### PR TITLE
implementation of the Facade pattern

### PR DESCRIPTION
**Pull request by andreykalachev**
Added VehicleToFileLogger that allows to save or retrieve vehicles to/from a file. This class implements Facade pattern

### CODE REVIEW

File: TenDesignPatterns/Models/Interfaces/IVehicle.cs [hunk 1]
```diff
@@ -10,6 +10,6 @@ public interface IVehicle
     {
         void Drive();
         void Stop();
-        void Info();
+        string Info();
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Models/Interfaces/IVehicle.cs]: rename Info -> Inform or GetInfo

File: TenDesignPatterns/Program.cs [hunk 2]
```diff
@@ -15,24 +16,39 @@ public static void FactoryVehiclesInfo(IAbstractVehicleFactory abstractFactory)
...
             var truck = abstractFactory.CreateTruck();
 
-            car.Info();
-            truck.Info();
+            Console.WriteLine(car.Info());
+            Console.WriteLine(truck.Info());
         } 
         static void Main(string[] args)
         {
+            // abstract factory usage
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Program.cs]: // Abstract factory usage - Start
...
// Abstract factory usage - End

File: TenDesignPatterns/Program.cs [hunk 3]
```diff
@@ -15,24 +16,39 @@ public static void FactoryVehiclesInfo(IAbstractVehicleFactory abstractFactory)
...
             var mercedesFactory = new MercedesFactory();
             FactoryVehiclesInfo(mercedesFactory);
             Console.WriteLine();
 
             var volvoFactory = new VolvoFactory();
             FactoryVehiclesInfo(volvoFactory);
+            // abstract factory usage
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Program.cs]: Leave only one **empty** line between code segments

File: TenDesignPatterns/Experiments.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+using System.Linq;
+using TenDesignPatterns.Factories;
+using TenDesignPatterns.Persistence;
+using TenDesignPatterns.Services;
+
+namespace TenDesignPatterns
+{
+    public static class Experiments
+    {
+        public static void Print(string s = null)
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Experiments.cs]: public static void Print(string s = string.Empty)
        {
            Console.WriteLine(s);
        }

What do you think about this approach?

Comment (andreykalachev) [reply TenDesignPatterns/Experiments.cs]: seems more logical to me

File: TenDesignPatterns/Program.cs [hunk 5]
```diff
@@ -1,38 +1,25 @@
...
 {
-    class Program
+    public class Program
     {
-        public static void FactoryVehiclesInfo(IAbstractVehicleFactory abstractFactory)
+        public static void Main(string[] args)
         {
-            var car = abstractFactory.CreateCar();
-            var truck = abstractFactory.CreateTruck();
+            // abstract factory usage - start
```
Comment (dmyhun-andersen) [inline TenDesignPatterns/Program.cs]: #region Abstract Factory Pattern
....
#endregion

that would be even better

**Pull request merged**